pkg/apiclient/rollout/rollout.proto

```diff
@@ -0,0 +1,179 @@
+  string icon = 2;
+  int32 revision = 3;
+  string status = 4;
+  string message = 5;
+  repeated ReplicaSetInfo replicaSets = 6;
+  repeated AnalysisRunInfo analysisRuns = 7;
+}
+
+message ReplicaSetInfo {
+  k8s.io.apimachinery.pkg.apis.meta.v1.ObjectMeta objectMeta = 1;
+  string status = 2;
+  string icon = 3;
+  int32 revision = 4;
+  bool stable = 5;
+  bool canary = 6;
+  bool active = 7;
+  bool preview = 8;
+  int32 replicas = 9;
+  int32 available = 10;
+  string template = 11;
+  string scaleDownDeadline = 12;
+  repeated string images = 13;
+  repeated PodInfo pods = 14;
+}
+
+message PodInfo {
+  k8s.io.apimachinery.pkg.apis.meta.v1.ObjectMeta objectMeta = 1;
+  string status = 2;
+  string icon = 3;
+  string ready = 4;
+  int32 restarts = 5;
+}
+
+message ContainerInfo {
+  string name = 1;
+  string image = 2;
+}
+
+message JobInfo {
+  k8s.io.apimachinery.pkg.apis.meta.v1.ObjectMeta objectMeta = 1;
+  string status = 2;
+  string icon = 3;
+}
+
+message AnalysisRunInfo {
+  k8s.io.apimachinery.pkg.apis.meta.v1.ObjectMeta objectMeta = 1;
+  string icon = 2;
+  int32 revision = 3;
+  string status = 4;
+  int32 successful = 5;
+  int32 failed = 6;
+  int32 inconclusive = 7;
+  int32 error = 8;
+  repeated JobInfo jobs = 9;
+}
+
+service RolloutService {
+    // Get returns a rollout
+    rpc GetRollout(RolloutQuery) returns (RolloutInfo) {
+        option (google.api.http).get = "/api/v1/rollout/{name}";
+    }
+
+    rpc WatchRollout(RolloutQuery) returns (stream RolloutInfo) {
+        option (google.api.http).get = "/api/v1/rollout/watch/{name}";
+    }
+
+    rpc ListRollouts(google.protobuf.Empty) returns (RolloutInfoList) {
+        option (google.api.http).get = "/api/v1/rollouts";
+    }
+
+    rpc WatchRollouts(google.protobuf.Empty) returns (stream RolloutWatchEvent) {
+        option (google.api.http).get = "/api/v1/rollouts/watch";
+    }
+
+    rpc GetNamespace(google.protobuf.Empty) returns (NamespaceInfo) {
+        option (google.api.http).get = "/api/v1/namespace";
+    }
+
+    rpc RestartRollout(RolloutQuery) returns (google.protobuf.Empty) {
+        option (google.api.http).get = "/api/v1/rollout/restart/{name}";
+    }
+
+    rpc PromoteRollout(RolloutQuery) returns (google.protobuf.Empty) {
+        option (google.api.http).get = "/api/v1/rollout/promote/{name}";
+    }
+
+    rpc AbortRollout(RolloutQuery) returns (google.protobuf.Empty) {
+        option (google.api.http).get = "/api/v1/rollout/abort/{name}";
+    }
+
+    rpc SetRolloutImage(SetImageQuery) returns (google.protobuf.Empty) {
+        option (google.api.http).get = "/api/v1/rollout/set/{rollout}/{container}/{image=**}/{tag}";
+    }
+
+    rpc UndoRollout(UndoQuery) returns (google.protobuf.Empty) {
+        option (google.api.http).get = "/api/v1/rollout/undo/{rollout}/{revision}";
+    }
+
+    rpc RetryRollout(RolloutQuery) returns (google.protobuf.Empty) {
+        option (google.api.http).get = "/api/v1/rollout/retry/{name}";
+    }
```

Comment:
1. All of these mutation API's which currently use the same `RolloutQuery` object, need to have individual Request objects. e.g.

* PromoteRolloutRequest
* AbortRolloutRequest
* UndoRolloutRequest
* RetryRolloutRequest

This is because each of these APIs potentially will have their own options, and we cannot reuse the RolloutQuery object to have all the options for each endpoint.

2. To follow REST semantics, these should be PUT requests instead of GET, with the subfield coming after the rollout name. e.g. 

```
option (google.api.http).put = "/api/v1/rollout/{name}/abort";
```

3. Instead of returning a `google.protobuf.Empty`, it would be better to return the updated Rollout object with the modifications.